File: app/src/main/java/com/jameskbride/eventbusmvpdemo/bus/BusAware.java
package com.jameskbride.eventbusmvpdemo.bus;

import org.greenrobot.eventbus.EventBus;

public class BusAware {

    protected EventBus bus;

    public BusAware(EventBus bus) {
        this.bus = bus;
    }

    public void open() {
        bus.register(this);
    }

    public void close() {
        bus.unregister(this);
    }
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/bus/BusAware.kt
package com.jameskbride.eventbusmvpdemo.bus

import org.greenrobot.eventbus.EventBus

interface BusAware {
    val eventBus: EventBus
    fun open() {
        eventBus.register(this)
    }

    fun close() {
        eventBus.unregister(this)
    }
}